ts/test/components/AvatarPlaceHolder_test.tsx

```diff
@@ -56,71 +46,33 @@ describe('AvatarPlaceHolder', () => {
+    expect(areResultsEqual(result, result2, true)).to.equal(true);
     result.unmount();
+    result2.unmount();
   });
-  it('should render the background color using the primary colors in the correct order', async () => {
-    const testPubkeys = [
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382fc', // green
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382fa', // blue
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382fd', // yellow
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382ff', // pink
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382ed', // purple
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382f9', // orange
-      '0541214ef26572066f0535140b1d6d021218299321c6001e2cdcaaa8cd5c9382eb', // red
-    ];
-
-    // NOTE we can trust the order of Object.keys and Object.values to be correct since our typescript build target is 'esnext'
-    const primaryColorKeys = Object.keys(COLORS.PRIMARY);
-    const primaryColorValues = Object.values(COLORS.PRIMARY);
-
-    async function testBackgroundColor(testPubkey: string, expectedColorValue: string) {
-      const result = renderComponent(
-        <AvatarPlaceHolder
-          diameter={AvatarSize.XL}
-          name={displayName}
-          pubkey={testPubkey}
-          dataTestId="avatar-placeholder"
-        />
-      );
-
-      // calculating the hash and initials needs to be done first
-      await waitFor(() => {
-        result.getByText('HW');
-      });
-
-      const el = result.getByTestId('avatar-placeholder');
-      const circle = el.querySelector('circle');
-      const circleColor = circle?.getAttribute('fill');
-      expect(circleColor, 'background color should not be null').to.not.equal(null);
-      expect(circleColor, 'background color should not be undefined').to.not.equal(undefined);
-      expect(circleColor, 'background color should not be an empty string').to.not.equal('');
-      expect(
-        primaryColorValues.includes(circleColor!),
-        'background color should be in COLORS.PRIMARY'
-      ).to.equal(true);
-      expect(
-        circleColor,
-        `background color should be ${primaryColorKeys[primaryColorValues.indexOf(expectedColorValue)]} (${expectedColorValue}) and not ${primaryColorKeys[primaryColorValues.indexOf(circleColor!)]} (${circleColor}) for testPubkey ${testPubkeys.indexOf(testPubkey)} (${testPubkey})`
-      ).to.equal(expectedColorValue);
-      result.unmount();
-    }
+  it('should render the background using a color from our theme', async () => {
+    const testPubkey = TestUtils.generateFakePubKeyStr();
+    const result = renderComponent(
+      // NOTE we need to test the pubkey to color generation and ordering with appium. Since we can't access the value of a css variable in with the current unit test setup
+      <AvatarPlaceHolder
+        diameter={AvatarSize.XL}
```

Comment:
can we not at least verify that the string var(--color-xxx) is the right one for each pubkeys?

Reply:
Unfortunately not because it possible from the component because the color is just the primary color variable. I wanted avoid abstrating the color generation into a function. Which is what strated unit testing components in the first place. I can do that and then we can test the individual color values.

i.e. The color prop returned is `var(--primar-color)` it does not resolve to a hex code